databricks/sdk/credentials_provider.py

```diff
@@ -277,6 +323,10 @@ def google_credentials(cfg: 'Config') -> Optional[HeaderFactory]:
     gcp_credentials = service_account.Credentials.from_service_account_info(info=account_info,
                                                                             scopes=GcpScopes)
 
+    def token() -> Token:
+        credentials.refresh(request)
```

Comment:
Do these need to be refreshed every time?

Reply:
Note that we do the same for normal credentials.

But in this case, I think it makes even more sense. We are going to cache the resulting Token, and the resulting Token has the same expiration as the original one. So refreshing it on creation makes sense so it will last the longest possible.